### PR TITLE
Add components to support text inputs with buttons inside

### PR DESCRIPTION
**Pull request by mperrotti**
## Screenshot
Screen Shot 2019-11-05 at 6 48 06 PM

### CODE REVIEW

File: packages/textInput/components/TextInputWithBadges.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
   inputContainer
 } from "../../shared/styles/formStyles";
 import { Badge } from "../../badge";
-import { flex, textTruncate } from "../../shared/styles/styleUtils";
+import { flex, textTruncate, flexItem } from "../../shared/styles/styleUtils";
```
Comment (mperrotti) [inline packages/textInput/components/TextInputWithBadges.tsx]: The changes in this file are not related to the text input with buttons. I happened to notice two bugs with this component I wanted to fix:
1. It didn't render icons
2. The text input would wrap to a new line when icons were present and the input had a narrow width

File: packages/textInput/components/TextInputWithIcon.tsx [hunk 2]
```diff
@@ -6,7 +6,8 @@ import { TextInput, TextInputProps } from "./TextInput";
 import {
   getInputAppearanceStyle,
   inputContainer,
-  getIconAppearanceStyle
+  getIconAppearanceStyle,
+  inputIconWrapper
```
Comment (mperrotti) [inline packages/textInput/components/TextInputWithIcon.tsx]: The changes in this file are not related to the text input with buttons.

I noticed that icons could be set to any size when passed into the input, and that leaves us open to visual inconsistency. `inputIconWrapper` limits the size

File: packages/textInput/README.md [hunk 3]
```diff
@@ -32,3 +32,15 @@ If the `tooltipContent` prop is set, an icon tooltip with the given text will be
 
 ## TextInputWithBadges
 `TextInputWithBadges` extends the `TextInputWithIcon` component. More info coming soon...
+
+## TextInputWithButtons
+`TextInputWithButtons` extends the `TextInputWithIcon` component, does not take an `iconEnd` prop.
+
+A `TextInputWithButtons` component are used when there is an action a user can take that is related to the text input. For example, clicking a button to clear the input's value.
```
Comment (GeorgiSTodorov) [inline packages/textInput/README.md]: I think it should be `is used`.
